src/playground/sample_data_ingestion/sample_data_ingestor/tests/unit-test/sample_data_ingestor_unit_test.py

```diff
@@ -0,0 +1,103 @@
+# opensearch-py==2.0.0
+from opensearchpy import OpenSearch
+
+# Standard libraries
+from json import loads, load
+import pytest
+import sys
+import os
+
+# Adds the directory "/sample_data_ingestion" to sys.path
+sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+from sample_data_ingestor import ingest
+
+# Constants
+INDEX_NAME = "ingest-test"
+DIR_PATH = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+INDEX_BODY = {
+  'settings': {
+    'index': {
+      'number_of_shards': 1
+    }
+  }
+}
+
+
+# Configurations for connection (NEED TO CHANGE)
+HOST = os.environ.get("HOST")
+PORT = 9200
+AUTH = (os.environ.get("SAMPLE_DATA_USERNAME"), os.environ.get("SAMPLE_DATA_PASSWORD"))
+
+
+# Establish connection with OS
+client = OpenSearch(
+    hosts = [{'host': HOST, 'port': PORT}],
+    http_compress = True,
+    http_auth = AUTH,
+    use_ssl = True,
+    verify_certs = False,
+    ssl_assert_hostname = False,
+    ssl_show_warn = False
+)
+
+
+# Sample inputs (valid)
+valid_test_inputs = {}
+with open(DIR_PATH + "/test-files/valid-template-inputs.json", "r") as f:
+    valid_test_inputs = load(f)
+
+jinput = valid_test_inputs["jinput"]
+
+jinput2 = valid_test_inputs["jinput2"]
+
+jinput3 = valid_test_inputs["jinput3"]
+
+jinput4 = valid_test_inputs["jinput4"]
+
+jinput5 = valid_test_inputs["jinput5"]
+
+jinput6 = valid_test_inputs["jinput6"]
+
+
+# Sample index for testing 
+if not client.indices.exists(index = INDEX_NAME):
+    client.indices.create(index = INDEX_NAME, body = INDEX_BODY)
```

Comment:
I'm not clear here, are you calling the "real" service? or mock here?

Reply:
I am calling the "real" service, since the `ingest()` function makes a `BULK` request.

Reply:
Latest commit now makes a "mock" call. I added a `DummyTransporter` to the `OpenSearch` client to test the functionality of ingest.